controller/src/main/java/io/pravega/controller/store/stream/tables/TableHelper.java

```diff
@@ -663,17 +663,23 @@ public static boolean isRerunOf(final List<Integer> segmentsToSeal,
     public static boolean isScaleInputValid(final List<Integer> segmentsToSeal,
                                             final List<AbstractMap.SimpleEntry<Double, Double>> newRanges,
                                             final byte[] segmentTable) {
-        boolean newRangesPredicate = newRanges.stream().noneMatch(x -> x.getKey() >= x.getValue() &&
-                x.getKey() >= 0 && x.getValue() > 0);
-
         List<AbstractMap.SimpleEntry<Double, Double>> oldRanges = segmentsToSeal.stream()
                 .map(segment -> SegmentRecord.readRecord(segmentTable, segment).map(x ->
                         new AbstractMap.SimpleEntry<>(x.getRoutingKeyStart(), x.getRoutingKeyEnd())))
                 .filter(Optional::isPresent)
                 .map(Optional::get)
                 .collect(Collectors.toList());
 
-        return newRangesPredicate && reduce(oldRanges).equals(reduce(newRanges));
+        if (newRanges.isEmpty()) {
+            // Stream is being sealed;
+            return reduce(oldRanges).get(0).equals(new AbstractMap.SimpleEntry<>(0.0, 1.0));
```

Comment:
should we want to check if segmentsToSeal is also empty in case seal is called idempotently.. 
it may have been checked elsewhere, but for correctness of this method, it could perform the validation or we may get NPE here..